pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -332,34 +322,72 @@ func (ia *IngressAnalyzer) getIngressObjectTargetedPeersAndPorts(ns string,
 	svcList []serviceInfo) (map[eval.Peer]*common.ConnectionSet, error) {
 	res := make(map[eval.Peer]*common.ConnectionSet)
 	for _, svc := range svcList {
-		peers, ok := ia.servicesToPeersMap[ns][svc.serviceName]
+		peersAndPorts, ok := ia.servicesToPortsAndPeersMap[ns][svc.serviceName]
 		if !ok {
 			ia.logger.Warnf("ignoring target service " + svc.serviceName + " : service not found")
 		}
-		for _, peer := range peers {
-			// if the service port is namedPort , check if the peer contains this port name and get its matching port number
-			portNum := svc.serviceTargetPort.IntValue()
-			if svc.serviceTargetPort.StrVal != "" {
-				portInt, err := ia.pe.ConvertPeerNamedPort(svc.serviceTargetPort.StrVal, peer)
-				if err != nil {
-					return nil, err
-				}
-				portNum = int(portInt)
+		for _, peer := range peersAndPorts.peers {
+			currIngressPeerConn, err := ia.getIngressPeerConnection(peer, peersAndPorts.ports, svc.servicePort)
+			if err != nil {
+				return nil, err
 			}
-			permittedPeerConn := common.MakeConnectionSet(false)
-			// check if its TCP conns contains the required port
-			peerTCPConn := eval.GetPeerExposedTCPConnections(peer)
-			if peerTCPConn.Contains(strconv.Itoa(portNum), string(corev1.ProtocolTCP)) {
-				permittedPort := common.PortSet{}
-				permittedPort.AddPort(intstr.FromInt(portNum))
-				permittedPeerConn.AddConnection(corev1.ProtocolTCP, permittedPort)
-				if _, ok := res[peer]; !ok {
-					res[peer] = permittedPeerConn
-				} else {
-					res[peer].Union(permittedPeerConn)
-				}
+			if _, ok := res[peer]; !ok {
+				res[peer] = currIngressPeerConn
+			} else {
+				res[peer].Union(currIngressPeerConn)
 			}
 		}
 	}
 	return res, nil
 }
+
+// getIngressPeerConnection returns the ingress connection to a peer based on the required port specified in the ingress objects
+func (ia *IngressAnalyzer) getIngressPeerConnection(peer eval.Peer, actualServicePorts []corev1.ServicePort,
+	requiredPort intstr.IntOrString) (*common.ConnectionSet, error) {
+	peerTCPConn := eval.GetPeerExposedTCPConnections(peer)
+	// if the required port is not specified - all peer's TCP connections are allowed for ingress
+	if requiredPort.IntVal == 0 && requiredPort.StrVal == "" {
+		return peerTCPConn, nil
+	}
```

Comment:
doesn't it depend on what the service exposes?

Reply:
true - will change

Reply:
updated code